analyzer/codechecker_analyzer/analyzers/clangsa/config_handler.py

```diff
@@ -11,14 +11,51 @@
 from __future__ import division
 from __future__ import absolute_import
 
+import re
+import subprocess
+
 from codechecker_common.logger import get_logger
 from .ctu_autodetection import CTUAutodetection
 
+from . import clang_options
+from . import version
+
 from .. import config_handler
 
 LOG = get_logger('analyzer.clangsa')
 
 
+def parse_checkers(clangsa_output):
+    """
+    Parse clang static analyzer checkers list output.
+    Return a list of (checker name, description) tuples.
+    """
+
+    # Checker name and description in one line.
+    pattern = re.compile(
+        r'^\s\s(?P<checker_name>\S*)\s*(?P<description>.*)')
+    checkers_list = []
+    checker_name = None
+    for line in clangsa_output.splitlines():
```

Comment:
I know this change is just moving code from one function to another, but this parsing logic would be much more understandable with some modifications:
1. extend the docstring to make the format of the output clear (output to be parsed contains checker name and description pairs, which however can extend over multiple lines), add the fact that this is why we need stateful parsing when iterating over lines
2. rename `pattern` to something more meaningful (like `checker_entry_pattern`), that signifies that this pattern signals the beginning of a new checker entry
3. make the match object of `'^\s\s\S'` pattern a named, precompiled one (with a name like `indented_pattern`, so we could write in the condition `if indented_pattern.match(line):`. This would increase readability.